src/hotspot/share/gc/g1/heapRegionRemSet.inline.hpp

```diff
@@ -123,10 +123,7 @@ void HeapRegionRemSet::split_card(OopOrNarrowOopStar from, uint& card_region, ui
 }
 
 void HeapRegionRemSet::add_reference(OopOrNarrowOopStar from, uint tid) {
-  RemSetState state = _state;
-  if (state == Untracked) {
-    return;
-  }
+  assert(_state != Untracked, "must be");
```

Comment:
I would prefer "precondition" in the assert msg.

Reply:
Thanks Albert for the suggestion, I kinda agree.
But as the nearby code at line 122 uses "must be", so I will keep it for this case.

Reply:
L122 is a general invariant while here is a precondition literally.